File: app/src/main/java/com/gmail/fattazzo/formula1world/news/reader/NewsReader.java
package com.gmail.fattazzo.formula1world.news.reader;

import com.gmail.fattazzo.formula1world.news.NewsSource;
import com.gmail.fattazzo.formula1world.news.objects.News;

import java.util.List;

/**
 * @author fattazzo
 *         <p/>
 *         date: 20/07/17
 */
public interface NewsReader {

    /**
     * Load all news for the source.
     *
     * @param source news source
     * @return news
     */
    List<News> loadNews(NewsSource source);
}

File: app/src/main/kotlin/com/gmail/fattazzo/formula1world/news/reader/NewsReader.kt
package com.gmail.fattazzo.formula1world.news.reader

import com.gmail.fattazzo.formula1world.news.NewsSource
import com.gmail.fattazzo.formula1world.news.objects.News

/**
 * @author fattazzo
 *
 *
 * date: 20/07/17
 */
interface NewsReader {

    /**
     * Load all news for the source.
     *
     * @param source news source
     * @return news
     */
    fun loadNews(source: NewsSource): List<News>
}
